--- a/Makefile	2017-01-15 19:46:13.000000000 +0100
+++ b/Makefile	2017-01-21 20:45:06.593745779 +0100
@@ -197,6 +197,35 @@
 	TARGET := $(TARGET_NAME)_libretro_$(platform).bc
 	STATIC_LINKING = 1
 
+# ARM
+else ifneq (,$(findstring armv,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+	fpic := -fPIC -fsigned-char
+	CFLAGS += -D_GNU_SOURCE=1
+	CC = gcc
+ifneq (,$(findstring cortexa8,$(platform)))
+	CFLAGS += -marm -mcpu=cortex-a8
+	ASFLAGS += -mcpu=cortex-a8
+else ifneq (,$(findstring cortexa9,$(platform)))
+	CFLAGS += -marm -mcpu=cortex-a9
+	ASFLAGS += -mcpu=cortex-a9
+endif
+	CFLAGS += -marm
+ifneq (,$(findstring neon,$(platform)))
+	CFLAGS += -mfpu=neon
+	ASFLAGS += -mfpu=neon
+	HAVE_NEON = 1
+endif
+ifneq (,$(findstring softfloat,$(platform)))
+	CFLAGS += -mfloat-abi=softfp
+	ASFLAGS += -mfloat-abi=softfp
+else ifneq (,$(findstring hardfloat,$(platform)))
+	CFLAGS += -mfloat-abi=hard
+	ASFLAGS += -mfloat-abi=hard
+endif
+	CFLAGS += -DARM
+
 # Windows
 else
 	TARGET := $(TARGET_NAME)_libretro.dll
